pkg/api/types.go

```diff
@@ -799,6 +799,22 @@ type NodeCondition struct {
 	Message            string            `json:"message,omitempty"`
 }
 
+type NodeAddressKind string
+
+// These are valid address type of node. NodeLegacyHostIP is used to transit from out-dated
+// HostIP field to NodeAddress.
+const (
+	NodeLegacyHostIP NodeAddressKind = "LegacyHostIP"
+	NodeHostName     NodeAddressKind = "Hostname"
```

Comment:
A separate list sounds a little cubersome; essentially, if someone wants multiple hostnames or one kind of ip, they can just iterate the list like:
```
for i in addresses:
  if i.kind == hostname:
     names = append(names, i)
```

My approach here is to remove HostIP in internal representation, and change it to NodeAddresses[api.NodeLegacyHostIP].  After we remove support for v1beta1/2, we remove the LegacyHostIP.  Any problem with this approache, I think this makes it eaiser to do the migration.  I'd be happy to add HostIP back if you think this doesn't work well.